lib/helpers.py

```diff
@@ -186,6 +186,29 @@ def uplift_report(self, marks_and_spend_df, attributions_df):
 
         return report_df
 
+    def remove_users_marked_as_control_and_test(self, marks_and_spend_df):
+        """
+        Remove users that have been added to both the test and control group.
+
+        :param marks_and_spend_df: Marks and spend dataframe
+        :type marks_and_spend_df: pandas.DataFrame
+
+        :return: Dataframe without users marked as both test and control
+        :rtype: pandas.DataFrame
+        """
+        user_to_ab_test_group_count = marks_and_spend_df[marks_and_spend_df['event_type'] == 'mark'] \
+            .groupby('user_id', as_index=False).agg({'ab_test_group': pd.Series.nunique })
```

Comment:
I think it is nicer to wrap it in () instead of using \\
Maybe like this:
```python
    user_to_ab_test_group_count = (marks_and_spend_df[marks_and_spend_df['event_type'] == 'mark']
                                   .groupby('user_id', as_index=False)
                                   .agg({'ab_test_group': pd.Series.nunique}))
```
Don't know what the more "pythonic" thing is though

Reply:
Hmm, parenthesis seem to be the recommended way. There isn't really an explanation why (other than it is in PEP 8 whatever that is). I didn't really think about just googled it and took the recommendation of the first result. Parentheses have a objective disadvantage as they'll produce a bigger diff:


```python
a = 1 \
  + 2 \

# can be extended as:
a = 1 \
  + 2 \
+ + 3 \

# The recommended style:

a = (1
  + 2)

# =>

a (1
~ +2
+ + 3)
```

Not that I wrote it in an extendable way here 😛